libfwupd/fwupd-client.c

```diff
@@ -570,25 +571,38 @@ fwupd_client_signal_cb(GDBusProxy *proxy,
 {
 	FwupdClientPrivate *priv = GET_PRIVATE(self);
 	g_autoptr(FwupdDevice) dev = NULL;
+	g_autoptr(GError) error = NULL;
 	if (g_strcmp0(signal_name, "Changed") == 0) {
 		g_debug("Emitting ::changed()");
 		g_signal_emit(self, signals[SIGNAL_CHANGED], 0);
 		return;
 	}
 	if (g_strcmp0(signal_name, "DeviceAdded") == 0) {
-		dev = fwupd_device_from_variant(parameters);
+		dev = fwupd_device_new();
+		if (!fwupd_codec_from_variant(FWUPD_CODEC(dev), parameters, &error)) {
+			g_warning("failed to convert FwupdDevice: %s", error->message);
```

Comment:
I think you should make all these warnings unique so that they're easier to track the call path if they come up.

Reply:
Fixed.